src/libcore/slice/rotate.rs

```diff
@@ -73,7 +72,8 @@ pub unsafe fn ptr_rotate<T>(mut left: usize, mid: *mut T, mut right: usize) {
     }
 
     let mut rawarray = MaybeUninit::<RawArray<T>>::uninit();
-    let buf = &mut (*rawarray.as_mut_ptr()).typed as *mut [T; 2] as *mut T;
+    let buf = &mut (*rawarray.as_mut_ptr()).typed as *mut [MaybeUninit<T>; 2]
```

Comment:
Might be worth adding a note that this is creating a reference to uninitialized data, cc https://github.com/rust-lang/rfcs/pull/2582 ?

Reply:
The code doesn't do that anymore. It's all pointer ops now.